plenum/common/txn_util.py

```diff
@@ -126,11 +126,48 @@ def get_req_id(txn):
 
 
 def get_digest(txn):
-    return txn[TXN_PAYLOAD][TXN_PAYLOAD_METADATA].get(TXN_PAYLOAD_METADATA_DIGEST, None)
+    metadata = txn[TXN_PAYLOAD][TXN_PAYLOAD_METADATA]
+
+    # If there is no digest in metadata then it was meant to be so
+    digest = metadata.get(TXN_PAYLOAD_METADATA_DIGEST, None)
+    if digest is None:
+        return None
+
+    # Digest means full digest only if there is also a payload digest
+    if TXN_PAYLOAD_METADATA_PAYLOAD_DIGEST in metadata:
+        return digest
+
+    # Trying to restore full digest
+    # WARNING: This can still fail in following cases:
+    # 1) original request contained just one signature as part of signatures field
+    # 2) request contained some additional fields needed by plugins
+
+    op = get_payload_data(txn).copy()
```

Comment:
I would prefer not to do this since
1) If this is a txn without a full_digest received during catchup, then it's not stored in `seqNoDb` on other nodes in any case, so let's be consistent 
2) This fix doesn't work in all the cases (see Warnings above)
3) This is just one-time fix making the code more complex

I think it's better to fix `addBatch` in `seqNoDb` to support `None` full digests.

Reply:
Done